cf/formula_terms.py

```diff
@@ -0,0 +1,2057 @@
+
+            * The standard name of the computed non-parametric
+              coordinates.
+
+            * The computed non-parametric coordinates in a
+              `DomainAncillary` construct.
+
+            * A tuple of the domain axis construct keys for the
+              dimensions of the computed non-parametric coordinates.
+
+            * A tuple containing the construct key of the vertical
+              domain axis. If the vertical axis does not appear in the
+              computed non-parametric coodinates then this an empty
+              tuple, instead.
+
+    '''
+    standard_name = 'atmosphere_hybrid_sigma_pressure_coordinate'
+
+    computed_standard_name = _computed_standard_name(g, standard_name,
+                                                     coordinate_reference)
+
+    # ----------------------------------------------------------------
+    # Get the formula terms and their contruct keys
+    # ----------------------------------------------------------------
+    coordinate_conversion = coordinate_reference.coordinate_conversion
+
+    ap_term = 'ap' in coordinate_conversion.domain_ancillaries()
+
+    if ap_term:
+        ap, ap_key = _domain_ancillary_term(g, standard_name,
+                                            coordinate_conversion,
+                                            'ap', default_to_zero,
+                                            strict, True)
+
+        a_key = None
+    else:
+        a, a_key = _domain_ancillary_term(g, standard_name,
+                                          coordinate_conversion, 'a',
+                                          default_to_zero, strict,
+                                          True)
+
+        p0, _ = _domain_ancillary_term(g, standard_name,
+                                       coordinate_conversion, 'p0',
+                                       default_to_zero, strict, False)
+
+        ap_key = None
+
+    b, b_key = _domain_ancillary_term(g, standard_name,
+                                      coordinate_conversion, 'b',
+                                      default_to_zero, strict, True)
+
+    ps, ps_key = _domain_ancillary_term(g, standard_name,
+                                        coordinate_conversion, 'ps',
+                                        default_to_zero, strict,
+                                        False)
+
+    # Get the axes of the non-parametric coordinates, putting the
+    # vertical axis in postition -1 (the rightmost position).
+    k_axis = _vertical_axis(g, ap_key, a_key, b_key)
+    computed_axes = g.get_data_axes(ps_key) + k_axis
+
+    # Insert a size one dimension to allow broadcasting over the
+    # vertical axis
+    if k_axis:
+        ps = ps.insert_dimension(-1)
+
+    # ----------------------------------------------------------------
+    # Compute the non-parametric coordinates
+    # ----------------------------------------------------------------
+    if ap_term:
```

Comment:
It's not compulsory to change but think it would be more readable and natural to instead have a single `with` content management block outside of the `if`/`else` with a simple `if`/`else` to work out what term to use in the variable setting there, e.g. something like:

```python
if ap_term:
    use_ap_term = ap
else:
    use_ap_term = a * p0
```

Reply:
I agree in principle, but I think that it in this case we wouldn't gain much,  as the `bounds_combination_mode` call arguments are different in each case .

Reply:
Ah, fair point, I didn't notice that! Thought the `with` statements were identical but didn't detect that one had `ap` and the other `a`. In that case ignore my comment here, sorry.